libtelemetry/src/main/java/com/mapbox/android/telemetry/AlarmSchedulerFlusher.java

```diff
@@ -28,7 +28,7 @@
   @Override
   public void register() {
     Intent alarmIntent = receiver.supplyIntent(requestCode);
-    pendingIntent = PendingIntent.getBroadcast(context, requestCode, alarmIntent, FLAG_CANCEL_CURRENT);
+    pendingIntent = PendingIntent.getBroadcast(context, requestCode, alarmIntent, FLAG_UPDATE_CURRENT);
```

Comment:
i'm looking at this change closely and after reading api docs i'm not sure why we need `FLAG_UPDATE_CURRENT` at all? We don't seem to be updating any extras with this pending intent...do you have any insights into why would it be potentially beneficial?

Reply:
The issue is that `FLAG_CANCEL_CURRENT` does not actually cancel the alarm on Samsung devices. What we do with this flag is it cancels the current alarm (if it exists) and then we update it. Update allows us to replace the alarm. If we just use no flag (0), we will have the same issue on Samsung and other devices now. We will keep generating new alarms and not getting rid of old ones. This way we are saying to the system to use only one alarm, not more. 

A possible alternative is to detect if a duplicate pending intent has been created using `NO_CREATE`flag and if so, delete that old one and then generate a new pending intent manually instead of letting the system handle it.